src/config.c

```diff
@@ -179,8 +317,21 @@ int initialize_config(const char *config_file)
 	ini_config = CNC_MALLOC(sizeof(config_t));
 	ini_config->postgres_config = CNC_MALLOC(sizeof(postgres_t));
 	ini_config->smtp_config = CNC_MALLOC(sizeof(smtp_t));
+	ini_config->general_config = CNC_MALLOC(sizeof(general_t));
+	ini_config->general_config->log_filepath = NULL;
 	ret = ini_parse(config_file, handler, ini_config);
 
+	if (ret != 0) {
+		return ret;
+	}
+	ret = construct_log_filepath();
+	if (ret != 0) {
+		return ret;
+	}
+
+	construct_log_filename(log_filepath);
+	log_file = fopen(log_filename, "a");
```

Comment:
From the man pages of `FILE *fopen(const char *pathname, const char *mode);`

```man
RETURN VALUE
       Upon  successful  completion  fopen(), fdopen(), and freopen() return a
       FILE pointer.  Otherwise, NULL is returned and errno is set to indicate
       the error.
```

Reply:
I dont understand what should I do with this information.

Reply:
That can't be.

Reply:
The `if (log_file != NULL)` check should be done here also, right after opening the file.